payu/models/fms.py

```diff
@@ -106,13 +133,25 @@ def collate(self):
 
             mnc_tiles[t_base].append(t_fname)
 
-        # If this is run interactively NCPUS is set in collate_cmd, otherwise
-        # the cpu_count will return the number of CPUs assigned to the PBS job
-        count = int(os.environ.get('NCPUS', multiprocessing.cpu_count()))
-        pool = multiprocessing.Pool(processes=count)
+        cpucount = int(collate_conf.get('ncpus', multiprocessing.cpu_count()))
+
+        if mpi:
+            # Default to one for mpi
+            nprocesses = int(collate_conf.get('threads', 1))
+        else:
+            nprocesses = int(collate_conf.get('threads', cpucount))
+
+        ncpusperprocess = int(cpucount/nprocesses)
+
+        if ncpusperprocess == 1 and mpi:
+            print("Warning: running collate with mpirun with a single processor")
+
+        pool = multiprocessing.Pool(processes=nprocesses)
```

Comment:
Do you want to use pools with the MPI version?  Probably OK but could have some unexpected issues.

Reply:
I've tested it and it works. It allows for the most flexibility. Can you a priori think of any issues that might arise? Then I could test for it.

Reply:
I can't think of any specific problem.  I am just thinking that we are adding a process inbetween the parent process launched by PBS and the pool subprocess which is launching `mpirun`, and wondered if the PBS server and MPI process were fully aware of each other.

It's all probably OK, just wondering how it all plays out, or if another scheduler would object here.

Reply:
I agree it is a bit of a dog's breakfast, but actually it is sort of easier and cleaner to do it this way for the moment. What is the option? A great big
```python
if mpi:
else:
```
conditional?

Reply:
No it's fine, lets leave it for now